engine-src/Game/LambdaHack/Client/UI/HandleHelperM.hs

```diff
@@ -325,13 +322,15 @@ placesFromState coplace ClientOptions{sexposePlaces} s =
               EM.insertWith addEntries pk (ES.singleton lid, 0, 1, 0) em
             f (PK.PExists pk) em =
               EM.insertWith addEntries pk (ES.singleton lid, 0, 0, 1) em
-        in EM.foldr' f EM.empty lentry
+        in EM.foldr' f EM.empty lentry  -- go through place entrances and depending on place add an entry for whether Entry/Around/Exists - effect being we're counting #s of each type
       insertZeros !em !pk _ = EM.insert pk (ES.empty, 0, 0, 0) em
-      initialPlaces | not sexposePlaces = EM.empty
-                    | otherwise = ofoldlWithKey' coplace insertZeros EM.empty
+      initialPlaces | not sexposePlaces = EM.empty                                -- if not exposing places, don't list them
+                    | otherwise = ofoldlWithKey' coplace insertZeros EM.empty     -- otherwise do
```

Comment:
actually, "if not debug-exposing places, don't list places that have no entries in the dungeon at all, otherwise do even though the stats will be zeros (which may be valuable warning!)"

Reply:
and the warning that some place kind never gets generated in a 1000-level dungeon (`make dig-lore`; will take a while in a `optimization: False` build) is the main motivation for this whole feature IIRC